template/common-parameter-meta.txt

```diff
@@ -0,0 +1,36 @@
+        memory_gb: "RAM to allocate for task"
+        disk_size_gb: "Disk space to allocate for task"
+        modify_memory_gb: "Add to or subtract from dynamic memory allocation. Default memory is determined by the size of the inputs."
+        modify_disk_size_gb: "Add to or subtract from dynamic disk space allocation. Default disk size is determined by the size of the inputs."
+        ncpu: "Number of cores to allocate for task"
+        detect_nproc: "Use all available cores? Recommended for cloud environments. Not recommended for cluster environments."
+        max_retries: "<for a task>Number of times to retry in case of failure"
+        max_retries: "<for a workflow>Number of times to retry failed steps. Overrides task level defaults."
+        outfile_name: "Name for the <type of file> file"
+        prefix: "Prefix for the <type of file> file. The extension `<extension>` will be added."
+        gtf: "GTF features file"
```

Comment:
Right now, we use `gtf` everywhere when some tasks would technically allow a GFF. GTF is a stricter more specific form of GFF. Do we want to make that distinction? Or keep telling users "use a more specific file format"? Potentially more maintenance; to my knowledge we've never tested any of our workflows/task with a GFF, and we'd want to perform that sanity check before making a change to the documentation.

The easy option is keep specifying GTF format, if a user wants to use another type of GFF for whatever reason, they can figure it out themselves through trial and error.
That seems lazy and a bad user experience, so I'd suggest documenting (and testing) a wider set of formats.

Reply:
I don't have strong preferences on the file formats. That said, most (all?) of the major annotation sources provide annotations in both GTF and GFF format. So I would not go out of our way to ensure GFF compatibility. We would also need to be specific if we mean GFF3 or GFF2.

Reply:
I agree that we shouldn't put in much work to ensure compatibility with the GFF spec. *But* if a task is currently able to handle GFF2 or GFF3 (without us having to make code changes), that should be documented. Only proposing doc changes. Good point on v2/v3 compatibility. Will need some testing.

Reply:
Another thought on GTFs: we've discussed this through slack, but bringing it up on the public PR.

In some places we assume a gzipped GTF, and in other places we accept either compressed or uncompressed. Those cases should have separate documentation. Or better yet, we just universally update our tasks to accept either compressed or uncompressed GTFs.
I'm hesitant to remove a feature, but I'd also consider dropping support for uncompressed reference files so we can be consistent (this would also be the easiest solution, we wouldn't have to think about the uncompressed case anymore). I wouldn't actually advocate for taking that approach, but thought it was worth mentioning as an option.

Reply:
Closing due to being out of scope for this PR